### PR TITLE
Update go control plane and proxy to head.

### PR DESCRIPTION
**Pull request by bianpengyuan**

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]
If you have a chance to pull from the latest again that would be great, it has a small performance improvement

**Review by lei-tang — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]

File: pilot/pkg/networking/core/v1alpha3/listener_builder.go [hunk 1]
```diff
@@ -103,6 +103,7 @@ func amendFilterChainMatchFromInboundListener(chain *listener.FilterChain, l *li
 }
 
 func isBindtoPort(l *listener.Listener) bool {
+	// nolint: staticcheck
```
Comment (bianpengyuan) [inline pilot/pkg/networking/core/v1alpha3/listener_builder.go:106]: @howardjohn want to make sure it is ok to simply skip lint check.

Comment (howardjohn) [reply pilot/pkg/networking/core/v1alpha3/listener_builder.go:106]: Yes I think this is fine, we are intentionally using this. And we have an issue around cleaning up deprecated field usage

**Comment by bianpengyuan** [comment]
Looks like proxy update is needed to fix the test failure.. https://github.com/istio/proxy/pull/3139

File: pilot/pkg/networking/core/v1alpha3/listener_builder.go [hunk 2]
```diff
@@ -331,12 +331,12 @@ func (lb *ListenerBuilder) buildVirtualOutboundListener(configgen *ConfigGenerat
...
-		Name:                                VirtualOutboundListenerName,
-		Address:                             util.BuildAddress(actualWildcard, uint32(lb.push.Mesh.ProxyListenPort)),
-		Transparent:                         isTransparentProxy,
-		HiddenEnvoyDeprecatedUseOriginalDst: proto.BoolTrue,
-		FilterChains:                        filterChains,
-		TrafficDirection:                    core.TrafficDirection_OUTBOUND,
+		Name:             VirtualOutboundListenerName,
+		Address:          util.BuildAddress(actualWildcard, uint32(lb.push.Mesh.ProxyListenPort)),
+		Transparent:      isTransparentProxy,
+		UseOriginalDst:   proto.BoolTrue,
```
Comment (bianpengyuan) [inline pilot/pkg/networking/core/v1alpha3/listener_builder.go:337]: @howardjohn actually, do we need a version check on this?

Comment (howardjohn) [reply pilot/pkg/networking/core/v1alpha3/listener_builder.go:337]: Its the same proto field right? Just different name?

If its the same proto field, we are good - I believe it is

Comment (bianpengyuan) [reply pilot/pkg/networking/core/v1alpha3/listener_builder.go:337]: Ah I see.. I think so, but @tbarrella to double check.

Comment (tbarrella) [reply pilot/pkg/networking/core/v1alpha3/listener_builder.go:337]: The field is the same on the wire yeah

**Review by jacob-delgado — APPROVED** [review]

**Comment by jacob-delgado** [comment]
The 1.9 release process includes updating this dependency. There's a istio/build -> istio/proxy -> istio/istio (proxy sha) update and this change that should go into 1.9. I'll try and help you keep track of this.

**Comment by jacob-delgado** [comment]
Waiting on https://github.com/istio/proxy/pull/3139 to merge and then the update to istio/proxy sha into istio/istio

File: tests/integration/telemetry/policy/testdata/ratelimitservice.yaml [hunk 3]
```diff
@@ -104,7 +104,7 @@ spec:
         app: ratelimit
     spec:
       containers:
-      - image: envoyproxy/ratelimit:v1.4.0
+      - image: envoyproxy/ratelimit:6f5de117 # 2021/01/08
```
Comment (bianpengyuan) [inline tests/integration/telemetry/policy/testdata/ratelimitservice.yaml:107]: @mandarjog  @kyessenov @douglas-reid  Looks like envoyproxy/ratelimit release is not relaly maintained. envoy has dropped v2 proto version support, but the latest release from https://github.com/envoyproxy/ratelimit/releases is still 1.4.0 which can only talk v2.

In the [rate limiting doc](https://preliminary.istio.io/latest/docs/tasks/policy-enforcement/rate-limit/#global-rate-limit), we mentioned nothing about envoy rate limit service. So I think it is fine to just ping an alpha release build in the test?

**Review by mandarjog — APPROVED** [review]
lgtm

**Pull request merged**